src/features/Slices/postSlise.ts

```diff
@@ -0,0 +1,61 @@
+/* eslint-disable */
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import * as postsApi from '../../api/posts';
+import { Post } from '../../types/Post';
+
+export interface PostsState {
+  items: Post[];
+  loaded: boolean;
+  hasError: boolean;
+}
+
+const initialState: PostsState = {
+  items: [],
+  loaded: false,
+  hasError: false,
+};
+
+export const fetchPosts = createAsyncThunk(
+  'posts/fetchPosts',
+  async (userId: number) => {
+    const posts = await postsApi.getUserPosts(userId);
+
+    return posts;
+  },
+);
+
+const postsSlice = createSlice({
+  name: 'posts',
+  initialState,
+  reducers: {
+    clearPosts: state => {
+      state.items = [];
+    },
+  },
+  extraReducers: builder => {
+    builder
+      .addCase(fetchPosts.pending, state => {
+        return {
+          ...state,
+          loaded: true,
+        };
+      })
+      .addCase(fetchPosts.fulfilled, (state, action) => {
+        return {
+          ...state,
+          items: action.payload,
+          loaded: false,
+        };
+      })
+      .addCase(fetchPosts.rejected, state => {
+        return {
+          ...state,
+          hasError: true,
+          loaded: true,
+        };
+      });
```

Comment:
In your `extraReducers`, you are returning a new state object in each `addCase` method. This is not necessary when using Redux Toolkit with Immer, which allows you to write 'mutating' logic in reducers because it internally uses the Immer library to handle data immutability. So, you can directly modify the state.